code/surface_area.py

```diff
@@ -0,0 +1,421 @@
+    return str(n_vertices)
+
+
+# Map hemisphere from structure name
+def detect_hemi(structure_name: str, filename: str) -> str:
+    struct_map = {
+        "CortexLeft": "L",
+        "CortexRight": "R"
+    }
+    if structure_name in struct_map:
+        return struct_map[structure_name]
+    # fallback to filename patterns
+    lower = filename.lower()
+    if "lh" in lower or "left" in lower or ".l." in lower:
+        return "L"
+    if "rh" in lower or "right" in lower or ".r." in lower:
+        return "R"
+    return "na"
+
+
+# Detect space from filename or folder
+def detect_space(path: Path) -> str:
+    candidates = ["D99", "MEBRAINS", "Yerkes19", "CIVETNMT"]
+    for c in candidates:
+        if c.lower() in path.lower():
+            return c
+    return "unknown"
+
+
+def compute_surface_area(input_gifti: Path, output_metric: Path) -> None:
+    """
+    Compute the surface area of a brain surface mesh using niwrap.
+
+    """
+    # test if file exists
+    if not input_gifti.exists():
+        print(f"✗ Input file does not exist: {input_gifti}")
+        return
+
+    wb.surface_vertex_areas(surface=str(input_gifti), metric=str(output_metric))
+    print(f"Surface area metric saved to {output_metric}")
+
+
+def validate_output_file_data(output_metric: Path) -> bool:
+    """
+    Validate the output metric file by checking specific data fields using nibabel.
+    """
+    try:
+        # Basic file existence check
+        if not output_metric.exists():
+            print("✗ Output file does not exist")
+            return False
+
+        file_size = (
+            output_metric.stat().st_size
+        )  # Fix: use .stat().st_size instead of .size
+        if file_size == 0:
+            print("✗ Output file is empty")
+            return False
+
+        print(f"✓ File exists with size: {file_size} bytes")
+
+        # Load the GIFTI file using nibabel
+        try:
+            gifti_img = nib.load(str(output_metric))
+            print("✓ File loaded successfully with nibabel")
+        except Exception as e:
+            print(f"✗ Failed to load file with nibabel: {e}")
+            return False
+
+        validation_results = {}
+
+        # Check file type
+        if isinstance(gifti_img, nib.gifti.gifti.GiftiImage):
+            validation_results["Type"] = True
+            print("✓ Type: GIFTI Metric file")
+        else:
+            validation_results["Type"] = False
+            print("✗ Type: Not a GIFTI file")
+
+        # Check number of data arrays (maps)
+        num_maps = len(gifti_img.darrays)
+        if num_maps == 1:
+            validation_results["Number of Maps"] = True
+            print(f"✓ Number of Maps: {num_maps}")
+        else:
+            validation_results["Number of Maps"] = False
+            print(f"✗ Number of Maps: Expected 1, got {num_maps}")
+
+        # Check if it has data arrays (should map to surface)
+        if gifti_img.darrays and len(gifti_img.darrays) > 0:
```

Comment:
I think this was true when we were using workbench to check the file information. When we are checking the arrays, we don't know for certain this is mapped to a surface.